core/amber/src/main/scala/edu/uci/ics/texera/web/service/ExecutionResultService.scala

```diff
@@ -241,7 +245,7 @@ class ExecutionResultService(
               if (StorageConfig.resultStorageMode.toLowerCase == "mongodb") {
                 val opStorage = ResultStorage
                   .getOpResultStorage(workflowIdentity)
-                  .get(physicalPlan.getPhysicalOpsOfLogicalOp(opId).head.id.logicalOpId)
+                  .get(OpResultStorage.storageKey(opId, PortIdentity()))
```

Comment:
Add some comments here since this is expecting only the first port.

Reply:
done